struct.c

```diff
@@ -413,18 +413,16 @@ struct_make_members_list(va_list ar)
 {
     char *mem;
     VALUE ary, list = rb_ident_hash_new();
-    st_table *tbl = RHASH_TBL_RAW(list);
 
-    RBASIC_CLEAR_CLASS(list);
```

Comment:
I think you want to keep the CLEAR_CLASS to hide the object from ObjectSpace.each_object

Reply:
We were actually questioning the usefulness of it. It's frozen, is it a big deal if you can find it though ObjectSpace?

Reply:
Sorry, this one isn't frozen, but ephemeral, we don't keep a reference on it.

Reply:
I like to be conservative with this stuff. Not necessarily against removing it, but please at least do it in a separate commit. This is a separate concern from OBJ_WB_UNPROTECT.

Reply:
Ok, added back.